scrapy/extensions/logstats_extended.py

```diff
@@ -0,0 +1,132 @@
+import logging
+from datetime import datetime
+
+from twisted.internet import task
+
+from scrapy import signals
+from scrapy.exceptions import NotConfigured
+from scrapy.utils.serialize import ScrapyJSONEncoder
+
+logger = logging.getLogger(__name__)
+
+
+class LogStatsExtended:
+    """Log basic scraping stats periodically"""
+
+    def __init__(
+            self, stats, interval=60.0,
+            ext_stats_enabled=False,
+            ext_stats_include=None,
+            ext_stats_exclude=None,
+            ext_delta_enabled=False,
+            ext_delta_include=None,
+            ext_delta_exclude=None,
+            ext_timing_enabled=False,
+            ext_timing_include=None,
+            ext_timing_exclude=None,
+    ):
+        self.stats = stats
+        self.interval = interval
+        self.multiplier = 60.0 / self.interval
+        self.task = None
+        self.encoder = ScrapyJSONEncoder(sort_keys=True, indent=4)
+        self.ext_stats_enabled = ext_stats_enabled
+        self.ext_stats_include = ext_stats_include
+        self.ext_stats_exclude = ext_stats_exclude
+        self.ext_delta_enabled = ext_delta_enabled
+        self.ext_delta_include = ext_delta_include
+        self.ext_delta_exclude = ext_delta_exclude
+        self.ext_timing_enabled = ext_timing_enabled
+        self.ext_timing_include = ext_timing_include
+        self.ext_timing_exclude = ext_timing_exclude
+
+
+    @classmethod
+    def from_crawler(cls, crawler):
+        interval = crawler.settings.getfloat("LOGSTATS_INTERVAL")
```

Comment:
Instead of having 3 new `_ENABLED` settings, we could have a single one called `LOGSTATS_FOO`, where `FOO` would be whatever the prefix/suffix of the new class is, e.g. `LOGSTATS_EXTENDED` if we keep the current name.

```suggestion
        settings = crawler.settings
        interval = settings.getfloat("LOGSTATS_INTERVAL")
        if not interval or not settings.getfloat("LOGSTATS_EXTENDED"):
            raise NotConfigured
```

Then, to get those `_enabled` booleans, you could do:

```suggestion
        ext_stats_include = crawler.settings.getlist("LOGSTATS_STATS_INCLUDE", [])
        ext_stats_exclude = crawler.settings.getlist("LOGSTATS_STATS_EXCLUDE", [])
        ext_stats_enabled = bool(ext_stats_include) or bool(ext_stats_exclude)
```

It probably makes sense to move the `_enabled` definition later on, to `__init__`, when following this approach.

Reply:
> Instead of having 3 new _ENABLED settings, we could have a single one called LOGSTATS_FOO, where FOO would be whatever the prefix/suffix of the new class is, e.g. LOGSTATS_EXTENDED if we keep the current name.

I also have thoughts in this direction to make it work with.. 3 settings only, like this:
```
        "PERIODIC_LOG_STATS" : {
            "include": ["downloader/", "scheduler/", "log_count/", "item_scraped_count"],
            "exclude": ["scheduler/"]
        },

        "PERIODIC_LOG_DELTA": {
            "include": ["downloader/"]
        },
        "PERIODIC_LOG_TIMING_ENABLED" : True,
```
Where expected to interpret delta/stats as enabled on any non `None` setting value
It will look (conceptually) closer to conf of `FEEDS` setting